server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2126,6 +2126,11 @@ public boolean finalizeVirtualMachineProfile(final VirtualMachineProfile profile
             }
         }
 
+        String routerLogrotateFrequency = RouterLogrotateFrequency.valueIn(router.getDataCenterId());
+        s_logger.debug(String.format("The setting [%s] with value [%s] for the zone with UUID [%s], will be used to configure the logrotate service frequency" +
+                        " on the virtual router.", RouterLogrotateFrequency.key(), routerLogrotateFrequency, dc.getUuid()));
+        buf.append(String.format(" logrotatefrequency=%s", routerLogrotateFrequency));
```

Comment:
@RodrigoDLopez 
is there a check on the value so that the logrotate will not be broken due to invalid value ?

Reply:
I did not test with nonsense values, just the value of `null`, in this case, the default value will be applied and some values that make any sense, so I am not sure if a nonsense value will break logrotate service.
any suggestion to ensure that the operators will follow the pattern?

Reply:
@weizhouapache do you mean an invalid `logrotatefrequency`, and add a validator method for that?

Reply:
@DaanHoogland yes

@RodrigoDLopez 
have you tested with a random string ?

Reply:
@weizhouapache 
I was able to confirm that even when the operator enters a meaningless value, the Logrotate service continues to run. However, I was not able to identify which frequency value was assumed in this context. For this reason, a new method was implemented to check if the value provided makes sense, otherwise, we will use the default value. Can you please review it again?